src/status_im2/contexts/shell/events.cljs

```diff
@@ -29,59 +29,78 @@
 
 ;; Events
 
-(rf/defn add-switcher-card
-  {:events [:shell/add-switcher-card]}
-  [{:keys [db now] :as cofx} view-id id]
+(rf/defn switcher-cards-loaded
+  {:events [:shell/switcher-cards-loaded]}
+  [{:keys [db]} loaded-switcher-cards]
+  {:db (assoc db
+              :shell/switcher-cards
+              (reduce (fn [acc switcher-card]
```

Comment:
You can replace the entire reduce processing with status-im.utils.core/index-by.

Reply:
now, also renaming keys. So index-by will not be sufficient

Reply:
@Parveshdhull, you can still use `index-by` because the collection can be transformed outside the reducing function.

```clojure
(utils.core/index-by :card-id (switcher-cards-store/<-rpc loaded-switcher-cards))
```

Reply:
done using `clojure.walk`, thank you